src/libs/location/src/character.cpp

```diff
@@ -2828,21 +2931,48 @@ if(storm::iEquals(eventName, "Blade to belt"))
                 else if (storm::iEquals(eventName, CHARACTER_FIGHT_GUNHAND))
                 {
                     core.Send_Message(blade, "l", MSG_BLADE_GUNHAND);
+                    isFireState = true;
                 }
                 else if (storm::iEquals(eventName, CHARACTER_FIGHT_GUNFIRE))
                 {
+                    isFireState = false;
                     core.Send_Message(blade, "l", MSG_BLADE_GUNFIRE);
                     // PlaySound("pistol_shot");
                     isFired = true;
                     float kDist;
                     Character *chr = FindGunTarget(kDist, CheckShotOnlyEnemyTest());
                     entid_t enemy{};
+                    int32_t resHit = 0;
```

Comment:
I guess it might be a good idea to use enums instead simple values
@espkk what do you think about it?

Reply:
Fixed. Now resHit is a standard boolean with {0; 1} value.

Reply:
Maybe then it should be boolean, not int?

Reply:
also I suppose that `isEnemyFinded` souns better than `resHit `

Reply:
https://github.com/storm-devs/storm-engine/blob/f6c8caad16b4126257a082d53345d3673b5100f8/src/libs/location/src/character.cpp#L4981
https://github.com/storm-devs/storm-engine/blob/f6c8caad16b4126257a082d53345d3673b5100f8/src/libs/location/src/character.cpp#L4982

Check this out
`isGunBlade` here is bool

Reply:
or even add two bools for better readability
`isEnemyFound` for `Location_CharacterFire` and `isEnemyHitFire` for `Check_ChrHitFire `

Reply:
> Maybe then it should be boolean, not int?

> also I suppose that `isEnemyFound` sounds better than `resHit`

Sounds good. In my mod I use three different values of this var, and use `resHit` due to inertia(

> or even add two bools for better readability
`isEnemyFound` for `Location_CharacterFire` and `isEnemyHitFire` for `Check_ChrHitFire`

What is the first bool?
Btw I use int to get boolean from script according to the general codebase.
Like `tmpBool` in this function, for example:
![image](https://user-images.githubusercontent.com/110747804/192059770-3ff71018-e85d-4d82-ba05-58a1620e63e7.png)

Reply:
Hmmmm, nevermind about bool
I'm suggesting something like this:

```
                    entid_t enemy{};
                    int32_t isEnemyHitByGunfire = 1;

                    if (chr)
                    {
                        enemy = chr->GetId();
                        VDATA *vd = core.Event("Check_ChrHitFire", "iilf", GetId(), enemy,
                                               static_cast<int32_t>(chr->isRecoilState), kDist);
                        if (vd)
                        {
                            vd->Get(isEnemyHitByGunfire);
                        }

                        if (isEnemyHitByGunfire)
                        {
                            chr->Hit(fgt_hit_fire);
                        }
                    }
                    core.Event("Location_CharacterFire", "iifl", GetId(), enemy, kDist,
                               chr != nullptr && isEnemyHitByGunfire);
```

Reply:
I don't like operand `chr != nullptr`. You guess it's necessary?

Reply:
Yep, I think so

Reply:
Done.